sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -343,4 +385,139 @@ export class AssessmentSequentialProcessingService {
+        startDate: programYearStartDate,
+      })
+      .andWhere("sfasApplication.endDate < :endDate", {
+        endDate: firstAssessmentDate,
+      });
+
+    const awards =
+      await sfasApplicationAwards.getRawOne<SFASApplicationAwards>();
+    const totals: AwardTotal[] = [];
+    if (awards) {
+      Object.entries(awards).forEach(([key, value]) => {
+        if (+value > 0) {
+          totals.push({
+            offeringIntensity: OfferingIntensity.fullTime,
+            valueCode: key,
+            total: +value,
+          });
+        }
+      });
+    }
+    return totals;
+  }
+
+  /**
+   * Get SFAS part time (SAIL) application awards totals.
+   * @param assessment Current assessment being processed.
+   * @param firstAssessmentDate Date of the first assessment date of the current application.
+   * @returns SFAS application part time awards totals.
+   */
+  async getProgramYearSAILAwardsTotals(
+    assessment: StudentAssessment,
+    firstAssessmentDate: string,
+  ): Promise<AwardTotal[]> {
+    const lastName = assessment.application.student.user.lastName;
+    const sin = assessment.application.student.sinValidation.sin;
+    const birthDate = assessment.application.student.birthDate;
+    const programYearStartDate = assessment.application.programYear.startDate;
+    const sailApplicationAwards = this.sfasPartTimeApplicationsRepo
+      .createQueryBuilder("sfasPTApplication")
+      .select("SUM(sfasPTApplication.csgp_award)", "CSGP")
+      .addSelect("SUM(sfasPTApplication.sbsd_award)", "SBSD")
+      .addSelect("SUM(sfasPTApplication.csgd_award)", "CSGD")
+      .addSelect("SUM(sfasPTApplication.bcag_award)", "BCAG")
+      .addSelect("SUM(sfasPTApplication.cspt_award)", "CSPT")
+      .innerJoin("sfasPTApplication.individual", "sfasStudent")
+      .where("lower(sfasStudent.lastName) = lower(:lastName)", { lastName })
+      .andWhere("sfasStudent.sin = :sin", { sin })
+      .andWhere("sfasStudent.birthDate = :birthDate", { birthDate })
+      .andWhere("sfasPTApplication.startDate >= :startDate", {
+        startDate: programYearStartDate,
+      })
+      .andWhere("sfasPTApplication.endDate < :endDate", {
+        endDate: firstAssessmentDate,
+      });
+    const awards =
+      await sailApplicationAwards.getRawOne<SAILApplicationAwards>();
+    const totals: AwardTotal[] = [];
+    if (awards) {
+      Object.entries(awards).forEach(([key, value]) => {
+        if (+value > 0) {
+          totals.push({
+            offeringIntensity: OfferingIntensity.partTime,
+            valueCode: key,
+            total: +value,
+          });
+        }
+      });
+    }
+    return totals;
+  }
+
+  /**
+   * Merge disbursement totals, SFAS and SAIL application awards totals.
+   * @param disbursementTotals disbursement awards totals.
+   * @param sfasAwardsTotals SFAS application awards totals.
+   * @param sailAwardsTotals SAIL application awards totals.
+   */
+  mergeProgramYearAwardsTotals(
```

Comment:
I would say we can get rid of this whole method just but retuning the merged arrays from 

[simsAwardTotals , sfasAwardTotals, sfasPTAwardTotals]

STEP 1: 
![image](https://github.com/bcgov/SIMS/assets/54600590/ad72f079-ab48-4280-89dd-2065adfa3dda)

STEP 2:

![image](https://github.com/bcgov/SIMS/assets/54600590/ccecd17f-ac07-42f8-a7ad-71d3c1787c28)

STEP 3: 
![image](https://github.com/bcgov/SIMS/assets/54600590/532a5296-2571-4c66-9849-4149f5a20099)

Reply:
This type of operation is perfect for unit tests also. If we have E2E testing covering the scenarios there is no need to add unit tests. But in one way or another, we should ensure some coverage.